File: src/main/java/info/rsdev/playlists/spotify/TooManyRequestsExceptionHandler.java
package info.rsdev.playlists.spotify;

import com.wrapper.spotify.exceptions.detailed.TooManyRequestsException;
import org.slf4j.Logger;

public abstract class TooManyRequestsExceptionHandler {
	
	private TooManyRequestsExceptionHandler() {};
	
    public static void handle(Logger log, String requestName, TooManyRequestsException e) {
    	int sleepSecs = e.getRetryAfter() + 1;
    	log.warn(String.format("%s: API Rate limit exceeded. Going to sleep for %d seconds", requestName, sleepSecs));
		sleep(e.getRetryAfter() + 1);
		log.warn("Waking up and going on");
    }

    public static void handle(TooManyRequestsException e) {
		sleep(e.getRetryAfter() + 1);
    }

    private static void sleep(int sleepSecs) {
		try {
			Thread.sleep(sleepSecs * 1000L);
		} catch (InterruptedException ie) {
			Thread.currentThread().interrupt();
		}
    }

}

File: src/main/kotlin/info/rsdev/playlists/spotify/TooManyRequestsExceptionHandler.kt
package info.rsdev.playlists.spotify

import com.wrapper.spotify.exceptions.detailed.TooManyRequestsException
import org.slf4j.Logger

object TooManyRequestsExceptionHandler {

    fun handle(log: Logger, requestName: String, e: TooManyRequestsException) {
        val sleepSecs = e.retryAfter + 1
        log.warn("$requestName: API Rate limit exceeded. Going to sleep for $sleepSecs seconds")
        sleep(sleepSecs)
        log.warn("Waking up and going on")
    }

    private fun sleep(sleepSecs: Int) {
        try {
            Thread.sleep(sleepSecs * 1000L)
        } catch (ie: InterruptedException) {
            Thread.currentThread().interrupt()
        }
    }

}
